packages/common/src/dto-types.ts

```diff
@@ -76,16 +77,26 @@ export class UpdatePlanDto {
   warnings?: IWarning[];
 }
 
-export class SignUpDto {
-  @IsNotEmpty()
-  @IsEmail()
+export class SignUpStudentDto {
+  @IsNotEmpty({ message: "Email cannot be empty" })
+  @IsEmail({ message: "Email must be an email" })
   email: string;
 
-  @IsNotEmpty()
-  @IsString()
+  @IsNotEmpty({ message: "Password cannot be empty" })
+  @IsString({
+    message: "Password must only contain alphanumeric values and signs",
+  })
   password: string;
-}
 
+  @IsNotEmpty({ message: "Confirm Password cannot be empty" })
+  @IsString({
+    message: "Confirm Password must only contain alphanumeric values and signs",
+  })
+  @IsEqualTo("password", {
+    message: "Confirm Password must be same as Password",
+  })
+  passwordConfirm: string;
```

Comment:
sorry if u put a lot of work into the decorator thing, but i think more complex validation (that is not essentially jsut property checking) should be put in the actual api

Reply:
I'm not sure if I understand this correctly, but NestJS uses class-validator to perform its body validation. So our API actually uses the properties and decorators of class-validator to perform validation. Do you mean we instead switch to having the auth controller directly check if `confirmPassword == password`?

Reply:
yes, that's what i meant

Reply:
i think for simple validation like typechecking and the builtins that allow us to do value checks, it's good to have nest do the heavy lifting. but this specific annotation will probably never be used in another situation, so it's not worth the extra complexity of abstracting it